ui/packages/consul-ui/app/services/ui-config.js

```diff
@@ -1,14 +1,41 @@
 import Service, { inject as service } from '@ember/service';
 import { get } from '@ember/object';
 
+import dataSource from 'consul-ui/decorators/data-source';
+
 export default class UiConfigService extends Service {
   @service('env') env;
 
-  async findByPath(path, configuration = {}) {
-    return get(this.get(), path);
+  @dataSource('/:partition/:nspace/:dc/ui-config/:path')
+  async findByPath(params) {
+    return get(this.get(), params.path);
+  }
+
+  @dataSource('/:partition/:nspace/:dc/notfound/:path')
+  async parsePath(params) {
+    return params.path.split('/').reduce((prev, item, i) => {
+      switch (true) {
+        case item.startsWith('~'):
+          prev.nspace = item.substr(1);
+          break;
+        case item.startsWith('-'):
+          prev.partition = item.substr(1);
+          break;
+        case typeof prev.dc === 'undefined':
+          prev.dc = item;
+          break;
+      }
+      return prev;
+    }, {});
+  }
```

Comment:
All of this stuff is now so much easier to think about (the stuff we removed was in the application error hook)